rust/sealevel/programs/hyperlane-sealevel-token-collateral/src/plugin.rs

```diff
@@ -444,7 +444,7 @@ impl HyperlaneSealevelTokenPlugin for CollateralPlugin {
             vec![
                 AccountMeta::new_readonly(token.plugin_data.spl_token_program, false).into(),
                 AccountMeta::new_readonly(spl_associated_token_account::id(), false).into(),
-                AccountMeta::new(token.plugin_data.mint, false).into(),
+                AccountMeta::new_readonly(token.plugin_data.mint, false).into(),
```

Comment:
doesn this apply to synthetic routes as well? If so, worth changing there too even if those haven't caused issues

Reply:
Because a transfer out for a synthetic involves minting the token and increasing the total supply, the mint needs to be writeable :(

https://github.com/solana-labs/solana-program-library/blob/5380b0e7e2f4a479a450c1b7091690ddb312ba73/token/program-2022/src/instruction.rs#L1722-L1736